internal/core/types.go

```diff
@@ -112,6 +115,27 @@ type NewURLPart struct {
 
 func (p NewURLPart) isNewPart() {}
 
+// NewAttachmentPart has no JSON annotations because it cannot be sent directly
+// to the backend. The attachment must first be uploaded and a
+// newAttachmentPartUploaded sent instead.
+type NewAttachmentPart struct {
+	Type       string
+	Name       *string
+	CustomData interface{}
+	File       io.Reader
+}
+
+func (p NewAttachmentPart) isNewPart() {}
+
+type newAttachmentPartUploaded struct {
```

Comment:
`newAttachmentPartUploaded` or `newAttachmentPartUpload`, or indeed just `newAttachmentPart`?

I find the name of the type matching the action is represents having happened reasonable, *until* it is then the serialisation type for a future request. Or maybe that does make sense too.

Reply:
I wasn't going for the action so much as the state it's in. "This is a new attachment part which has been uploaded." And you can only use an attachment part which has been uploaded when making your final request, it would be an error to use `NewAttachmentPart` (the `Uploaded` is still key here).